apps/emqx_bridge/src/emqx_bridge_api.erl

```diff
@@ -985,9 +985,13 @@ call_operation(NodeOrAll, OperFunc, Args = [_Nodes, BridgeType, BridgeName]) ->
         {error, timeout} ->
             ?SERVICE_UNAVAILABLE(<<"Request timeout">>);
         {error, {start_pool_failed, Name, Reason}} ->
-            ?SERVICE_UNAVAILABLE(
-                bin(io_lib:format("Failed to start ~p pool for reason ~p", [Name, Reason]))
-            );
+            Msg = bin(io_lib:format("Failed to start ~p pool for reason ~p", [Name, Reason])),
+            case Reason of
+                nxdomain ->
```

Comment:
Nit. This sounds like a somewhat limited (and probably not 100% correct) way to distinguish between "this service doesn't really exist there" and "this service doesn't work". I doubt if it's even possible to answer it correctly. For example, you'd have to also consider `econnrefused` / `econnreset` and a large swath of other posix errors, somehow make sure that this `nxdomain` isn't transient, make sure that if you established a connection, then make sure the other side is indeed mysql or something that follows protocol correctly, etc.

Reply:
Indeed, I don't think it's possible to discern between those two cases.

The intention here was not to be comprehensive about all possible errors, but to give a starting point where we could distinguish between "there's a problem in EMQX" (i.e.: `ecpool` is crashing for unexpected reasons) and "there's a problem starting to the resource".

With that, then, we could extend that error handling for more cases as they arise, since, as you pointed out, there's a large swath of possible errors.  :sweat_smile:

Reply:
Got it. I'd then let exceptions (i.e. _crashing for unexpected reasons_) propagate to the piece of code that converts them into `500`s, and convert everything else (i.e. expected errors) here into `400`s / `422`s. Currently, it's really odd that we're trying to tell the API client that _our_ API is unavailable when we cannot reach bridge target. 🫠